src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1057,6 +1058,22 @@ MPI_Init(int *argc, char ***argv)
 	return rc;
 }
 
+int
+PMPI_Init(int *argc, char ***argv)
+{
+	int rc;
+
+	if (next_pmpi_init == NULL) {
+		next_pmpi_init = dlsym(RTLD_NEXT, "PMPI_Init");
```

Comment:
For my understanding, we are expecting to get the reference on the original PMPI_Init function.
However, I do understand how we are sure of which function reference we will get, if there is several LD_PRELOAD on this last one.  For example, if Darshan is redefining it, could it be possible that we get the reference on this last one.

Reply:
The issue of deadlock is caused by calling daos_init() inside MPI runtime library's MPI_Init()/PMPI_Init(). We could avoid this issue as long as the PMPI_Init() from libpil4dfs is executed before the PMPI_Init() implemented in MPI runtime library.  It does not not matter the PMPI_Init() from libpil4dfs or the PMPI_Init() from darshan is executed first.

Reply:
OK, thanks for the explanation.